pymongo/pool.py

```diff
@@ -1126,7 +1161,10 @@ def _reset(self, close):
                 self.operation_count = 0
             sockets, self.sockets = self.sockets, collections.deque()
             if close:
-                self.closed = True
+                self.state = CLOSED
+            # Clear the wait queue
+            self._max_connecting_cond.notify_all()
```

Comment:
So when this happens and there are threads waiting to create a connection, they will get notified (via either size_cond or max_connecting_cond) and end up raising an exception since the pool is not ready?